src/internet_identity/tests/integration/config.rs

```diff
@@ -2,189 +2,216 @@ use canister_tests::api::internet_identity as api;
-
-    assert_eq!(api::config(&env, canister_id)?, config);
-
-    let _ = upgrade_ii_canister_with_arg(&env, canister_id, II_WASM.clone(), None);
-
-    assert_eq!(api::config(&env, canister_id)?, config);
+    let canister_id = install_and_assert(&env, &config)?;
+    upgrade_ii_canister_with_arg(&env, canister_id, II_WASM.clone(), Some(config.clone()))?;
+    assert_config(&env, canister_id, &config)?;
 
     Ok(())
 }
 
 #[test]
-fn should_override_partially() -> Result<(), CallError> {
+fn should_update_related_origins() -> Result<(), CallError> {
     let env = env();
-    let related_origins = [
-        "https://identity.internetcomputer.org".to_string(),
-        "https://identity.ic0.app".to_string(),
-        "https://identity.icp0.io".to_string(),
-    ]
-    .to_vec();
-    let openid_google = Some(OpenIdConfig {
-        client_id: "https://example.com".into(),
-    });
-    let config = InternetIdentityInit {
-        assigned_user_number_range: Some((3456, 798977)),
-        archive_config: Some(ArchiveConfig {
-            module_hash: [17; 32],
-            entries_buffer_limit: 123789,
-            polling_interval_ns: 659871258,
-            entries_fetch_limit: 33,
-        }),
-        canister_creation_cycles_cost: Some(123),
-        register_rate_limit: Some(RateLimitConfig {
-            time_per_token_ns: 99,
-            max_tokens: 874,
-        }),
-        captcha_config: Some(CaptchaConfig {
-            max_unsolved_captchas: 788,
-            captcha_trigger: CaptchaTrigger::Dynamic {
-                threshold_pct: 12,
-                current_rate_sampling_interval_s: 456,
-                reference_rate_sampling_interval_s: 9999,
-            },
-        }),
-        related_origins: Some(related_origins),
-        openid_google: Some(openid_google),
+    let config_1 = InternetIdentityInit {
+        related_origins: Some(vec![
+            "https://identity.internetcomputer.org".into(),
+            "https://identity.ic0.app".into(),
+            "https://identity.icp0.io".into(),
+        ]),
+        ..DEFAULT_CONFIG
+    };
+    let config_2 = InternetIdentityInit {
+        related_origins: Some(vec!["https://identity.ic0.app".into()]),
+        ..DEFAULT_CONFIG
+    };
+    let disabled_config = InternetIdentityInit {
+        related_origins: Some(vec![]),
+        ..DEFAULT_CONFIG
     };
 
-    let canister_id = install_ii_canister_with_arg(&env, II_WASM.clone(), Some(config.clone()));
+    // Check if related origins are updated and then untouched
+    let canister_id_1 = install_and_assert(&env, &config_1)?;
+    upgrade_and_assert(&env, canister_id_1, &config_2, &config_2)?;
+    unrelated_change_and_assert(&env, canister_id_1)?;
 
-    assert_eq!(api::config(&env, canister_id)?, config);
+    // Check if related origins are disabled and then untouched
+    let canister_id_2 = install_and_assert(&env, &config_1)?;
+    upgrade_and_assert(&env, canister_id_2, &disabled_config, &disabled_config)?;
```

Comment:
Wouldn't it be better if the confi passed to `upgrade_and_assert` had mostly `None` fields except for the changes we are interested and then we expect the fields to be present from before?

If config logic substituted a `None` for some value, we wouldn't catch it here, right?